registry/storage/driver/swift/swift.go

```diff
@@ -0,0 +1,533 @@
+		if err != nil {
+			return bytesRead, parseError(path, err)
+		}
+
+		if n < chunkSize {
+			// We wrote all the data
+			if cursor+n < currentLength {
+				// Copy the end of the chunk
+				headers := make(swift.Headers)
+				headers["Range"] = "bytes=" + strconv.FormatInt(cursor+n, 10) + "-" + strconv.FormatInt(cursor+chunkSize, 10)
+				file, _, err := d.Conn.ObjectOpen(d.Container, d.swiftPath(path), false, headers)
+				if err != nil {
+					return bytesRead, parseError(path, err)
+				}
+				if _, err := io.Copy(currentSegment, file); err != nil {
+					return bytesRead, parseError(path, err)
+				}
+				file.Close()
+			}
+			if n > 0 {
+				currentSegment.Close()
+				bytesRead += n - max(0, offset-cursor)
+			}
+			break
+		}
+
+		currentSegment.Close()
+		bytesRead += n - max(0, offset-cursor)
+		multi = io.MultiReader(io.LimitReader(reader, chunkSize))
+		cursor += chunkSize
+		partNumber++
+	}
+
+	return bytesRead, nil
+}
+
+// Stat retrieves the FileInfo for the given path, including the current size
+// in bytes and the creation time.
+func (d *driver) Stat(ctx context.Context, path string) (storagedriver.FileInfo, error) {
+	info, _, err := d.Conn.Object(d.Container, d.swiftPath(path))
+	if err != nil {
+		return nil, parseError(path, err)
+	}
+
+	fi := storagedriver.FileInfoFields{
+		Path:    path,
+		IsDir:   info.ContentType == directoryMimeType,
+		Size:    info.Bytes,
+		ModTime: info.LastModified,
+	}
+
+	return storagedriver.FileInfoInternal{FileInfoFields: fi}, nil
+}
+
+// List returns a list of the objects that are direct descendants of the given path.
+func (d *driver) List(ctx context.Context, path string) ([]string, error) {
+	var files []string
+
+	prefix := d.swiftPath(path)
+	if prefix != "" {
+		prefix += "/"
+	}
+
+	opts := &swift.ObjectsOpts{
+		Prefix:    prefix,
+		Delimiter: '/',
+	}
+
+	objects, err := d.Conn.Objects(d.Container, opts)
+	for _, obj := range objects {
+		if !obj.PseudoDirectory {
+			files = append(files, "/"+strings.TrimSuffix(obj.Name, "/"))
+		}
+	}
+
+	return files, parseError(path, err)
+}
+
+// Move moves an object stored at sourcePath to destPath, removing the original
+// object.
+func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) error {
+	err := d.Conn.ObjectMove(d.Container, d.swiftPath(sourcePath),
+		d.Container, d.swiftPath(destPath))
+	if err != nil {
+		return parseError(sourcePath, err)
+	}
+
+	return nil
+}
+
+// Delete recursively deletes all objects stored at "path" and its subpaths.
+func (d *driver) Delete(ctx context.Context, path string) error {
+	opts := swift.ObjectsOpts{
+		Prefix: d.swiftPath(path),
+	}
+
+	objects, err := d.Conn.ObjectNamesAll(d.Container, &opts)
+	if err != nil {
+		return parseError(path, err)
+	}
+	if len(objects) == 0 {
+		return storagedriver.PathNotFoundError{Path: path}
+	}
+
+	for index, name := range objects {
+		objects[index] = name[len(d.Prefix):]
+	}
+
+	var multiDelete = true
```

Comment:
Drop the `multiDelete` boolean, make sure to check for the `Forbidden` error from `BulkDelete`:

```go
if d.BulkDeleteSupport {
  if _, err := d.Conn.BulkDelete(d.Container, objects); err != swift.Forbidden {
    return err
  }
}
```